### PR TITLE
#16276 - move superglobals access to controllers

### PR DESCRIPTION
**Pull request by MrMadClown**
### Description

As described in [` #16276`](https://github.com/phpmyadmin/phpmyadmin/issues/16276)

This PR aims to reduce the usage of SuperGlobals outside of Controllers. 
I will add more commits, but wanted to open the PR now, to get early feedback since this is my first Contribution in this Project.

### CODE REVIEW

File: libraries/classes/ReplicationGui.php [hunk 1]
```diff
@@ -70,9 +70,9 @@ public function getHtmlForErrorMessage(): string
      *
      * @return string HTML code
      */
-    public function getHtmlForPrimaryReplication(): string
+    public function getHtmlForPrimaryReplication(?bool $replClearScr, ?string $primaryAddUser): string
     {
-        if (! isset($_POST['repl_clear_scr'])) {
+        if ( is_null($replClearScr)) {
```
Comment (williamdes) [inline libraries/classes/ReplicationGui.php:75]: On this project we prefer to compare === null

**Review by williamdes — APPROVED** [review]
I like this work

**Review by MauricioFauth — APPROVED** [review]

**Comment by MauricioFauth** [comment]
Merged, thanks for your contribution!

File: libraries/classes/ReplicationGui.php [hunk 2]
```diff
@@ -240,19 +242,21 @@ public function getHtmlForReplicationChangePrimary($submitName): string
     /**
      * This function returns html code for table with replication status.
      *
-     * @param string $type     either primary or replica
-     * @param bool   $isHidden if true, then default style is set to hidden, default value false
-     * @param bool   $hasTitle if true, then title is displayed, default true
+     * @param ?string $connection   primary connection
```
Comment (williamdes) [inline libraries/classes/ReplicationGui.php:245]: I missed this but it is `null|string` to be phpdoc standard correct

**Pull request merged**